File: 2308-DivideArrayIntoEqualPairs/2308-DivideArrayIntoEqualPairs.py
# Last updated: 31/07/2025, 21:09:04
class Solution:
    def divideArray(self, nums: List[int]) -> bool:
        dic={}
        for i in (nums):
            if i in dic:
                dic[i]+=1
            else:
                dic[i]=1
        l=len(nums)
        while l>0:
            for key,value in dic.items():
                if value>=2:
                    dic[key]-=2
                    l-=2
                elif value==0:
                    continue    
                else:
                    return False
        return True            

        

File: 2308-DivideArrayIntoEqualPairs/2308-DivideArrayIntoEqualPairs.java
// Last updated: 31/07/2025, 21:08:59
class Solution {
    public boolean divideArray(int[] nums) {
        int[] arr=new int[501];
        for(int i=0;i<nums.length;i++)
        {
            arr[nums[i]]++;
        }
        for(int i=1;i<501;i++)
        {
            if(arr[i]%2==1)
            {
                return false;
            }
        }
        return true;
    }
}